services/gitlab/gitlab-pipeline-status.service.js

```diff
@@ -0,0 +1,108 @@
+'use strict'
+
+const Joi = require('joi')
+const validate = require('../../lib/validate')
+const BaseSvgScrapingService = require('../base-svg-scraping')
+const { InvalidParameter, NotFound } = require('../errors')
+const { isPipelineStatus } = require('./gitlab-helpers')
+
+const badgeSchema = Joi.object({
+  message: Joi.alternatives()
+    .try([isPipelineStatus, Joi.equal('unknown')])
+    .required(),
+}).required()
+
+const queryParamsSchema = Joi.object({
+  // TODO This accepts URLs with query strings and fragments, which should be
+  // rejected.
+  gitlab_url: Joi.string().uri({ scheme: ['https'] }),
+}).required()
+
+module.exports = class GitlabPipelineStatus extends BaseSvgScrapingService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      base: 'gitlab/pipeline',
+      format: '([^/]+)/([^/]+)(?:/([^/]+))?',
+      capture: ['user', 'repo', 'branch'],
+      // Trailing optional parameters don't work. The issue relates to the `.`
+      // separator before the extension.
+      // pattern: ':user/:repo/:branch?',
+      queryParams: ['gitlab_url'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Gitlab pipeline status',
+        urlPattern: ':user/:repo',
```

Comment:
Now that #2308 is in, lets use `pattern` in these 3 examples (I realise you submitted this PR before we merged that)

Reply:
Updated to used `pattern`, and `namedParams`, too.